### DISCUSSION TITLE
How to pass inherited python object back to cpp function correctly

### DISCUSSION THREAD

**Discussion by joeyskeys** [pybind/pybind11#4617] [Q&A]
I've defined a cpp class Vec and its corresponding trampoline class PyVec, just like the example from the document and it works fine.
And now there's a function that should takes Vec or perhaps its python inherited object as a parameter, how to make it work successfully on the virtual function(this function also has a python binding)?
A detailed example of the python part is like this:
```
class AnotherVec(Vec):
     def virtual_func(self):
            do_sth()

v = AnotherVec()
a_binding_func(v)
```

What I wanna do is to let the python user to define some callback function in python code and then be called in the cpp function.

**Comment by balcklive** [comment]
got the same question, do you solve it?
